### ISSUE TITLE
Training dataset processing issues.

### ISSUE THREAD

**Issue by xunmenglt** [weichow23/merit#2]
class PairwiseCLDataset(Dataset):
    """
    Dataset for pairwise contrastive learning, where each sample consists of a
    (query, positive_passage) pair. Negatives are sourced from other examples in the batch.
    """
    def __init__(self, data_path: str, processor, max_text_length: int = 1024):
        self.max_text_length = max_text_length
        self.processor = processor
        if isinstance(data_path, str):
            data_path = glob.glob(data_path)
            
        self.data = []
        for fp in data_path:
            with open(fp, 'r') as f:
                lines = [json.loads(line) for line in f]
                self.data.extend(lines)
            logger.info(f"Loaded {len(lines)} samples from {fp}")

    def __len__(self) -> int:
        return len(self.data)

    def __getitem__(self, idx: int) -> Dict:
        try:
            record = self.data[idx]
            
            # Prepare conversations for the query and the positive passage
            query_conv = prep_conv(record["query"])
            passage_conv = prep_conv(record["passage"][0])

            # Apply chat template
            query_texts = self.processor.apply_chat_template(query_conv, tokenize=False, add_generation_prompt=True)
            passage_texts = self.processor.apply_chat_template(passage_conv, tokenize=False, add_generation_prompt=True)

            # Process query
            image_inputs, video_inputs = process_vision_info(query_conv)
            query_inputs = self.processor(text=query_texts, images=image_inputs, videos=video_inputs, padding=True, return_tensors="pt")

            # Process passage
            image_inputs, video_inputs = process_vision_info(passage_conv)
            passage_inputs = self.processor(text=passage_texts, images=image_inputs, videos=video_inputs, padding=True, return_tensors="pt")

            return {
                "id": record["id"],
                "query": query_inputs,
                "passage": passage_inputs,
            }
        
        except Exception as e:
            logger.warning(f"Error processing item at index {idx}: {e}. Skipping and trying next item.")
            return self.__getitem__((idx + 1) % len(self))

------

Above is the Dataset Class I found for processing the dataset in the project, but the training data processed in this class is not consistent with the training data provided by the hf official website in the `queries-train.json` format. Could you please let me know if there are any related scripts for processing training data, so that I can replicate the results of the paper?

Is it the following process?

Before conversion:
```json
{
    "query": [
        1730852586410640613,
        1729573514286958445
    ],
    "pos_candidate": [
        1730852586410706149
    ],
    "hard_negative": [
        1730852586410771685
    ],
    "query instruction": "Find a product of shoes that have the same material with <Product 1> <image>\nWater-Resistant EVA Slippers with Slip-Resistant Sole for Comfortable Indoor and Outdoor Use </Product 1> and the same water resistance with <Product 2> <image>\nGloryQueen Ella Collection Women's PU Leather Open Toe Sandals with Slip-Resistant Flat Sole for Elegant Summer Comfort </Product 2>, featuring a unique color combination and adjustable strap design.",
    "idx": 0
}
```

After conversion:
```json
{
    "query": [
      {
        "role":"user",
        "content":[
          {"type": "image", "image": "path/to/1730852586410640613.jpg"},
          {"type": "image", "image": "path/to/1729573514286958445.jpg"},
          {"type": "text", "text": "Find a product of shoes that have the same material with <Product 1> <image>\nWater-Resistant EVA Slippers with Slip-Resistant Sole for Comfortable Indoor and Outdoor Use </Product 1> and the same water resistance with <Product 2> <image>\nGloryQueen Ella Collection Women's PU Leather Open Toe Sandals with Slip-Resistant Flat Sole for Elegant Summer Comfort </Product 2>, featuring a unique color combination and adjustable strap design."},
        ]
      }
    ],
    "passage":[
      [
        {
          "role":"system",
          "content":[
            {"type":"image","image":"path/to/1730852586410706149.jpg"},
            {"type": "text","text":"<image>\nimage title"}
          ]
        }
      ]
    ]
    "idx": 0
}
```
